src/wallet/bitcoin.ts

```diff
@@ -170,6 +168,23 @@ export class InMemoryBitcoinWallet implements BitcoinWallet {
     return "150";
   }
 
+  public async close(): Promise<any> {
+    const tasks = [];
+    if (this.pool.opened) {
+      tasks.push(await this.pool.close());
+    }
+
+    if (this.chain.opened) {
+      tasks.push(await this.chain.close());
+    }
+
+    if (this.walletdb.opened) {
+      tasks.push(await this.walletdb.close());
+    }
+
+    return Promise.all(tasks);
```

Comment:
The return value is not needed (we probably don't care anyways as we want to close everything). 
```suggestion
    await Promise.all(tasks);
```

if you would be doing 
```
return Promise.all(tasks);
```
the return value would be an array of all return values from the resolved promises above.

Reply:
Not sure about this one. It ultimately doesn't matter, but all the other async functions in this file return a promise, rather than awaiting at the end. For consistency I would prefer to keep it as it is.

With your change I believe JS awaits inside the function and then wraps the result inside a promise that will resolve immediately and returns this value. That feels worse to me.

Reply:
Actually, I disagree with myself now. Now that I specify the return type with https://github.com/comit-network/comit-js-sdk/pull/176/commits/be737e03bc8df0f8f59909dbb8cf499e269ad736 (which we definitely should do) it becomes annoying to later on compose the promises in a similar way.

Opting for awaiting at the end as you originally suggested, and returning `Promise<void>` instead.